wallet/src/account/output_cache/mod.rs

```diff
@@ -732,59 +745,131 @@ impl OutputCache {
+                        conflict_check.confirmed_account_nonce
+                    {
+                        if confirmed_tx.get_id() != tx.get_transaction().get_id()
+                            && uses_conflicting_nonce(
+                                unconfirmed_tx,
+                                confirmed_account,
+                                confirmed_account_nonce,
+                            )
+                        {
+                            conflicting_txs.insert(tx.get_transaction().get_id());
+                            continue;
+                        }
+                    }
                 }
             }
         }
 
-        conflicting_txs
-            .into_iter()
-            .map(|tx_id| self.txs.get(tx_id).expect("must be present"))
-            .collect_vec()
+        // Remove all descendants of conflicting txs
+        let mut conflicting_txs_with_descendants = vec![];
+
+        for conflicting_tx in conflicting_txs {
+            let txs_to_rollback = self.remove_from_unconfirmed_descendants(conflicting_tx);
+
+            // Mark conflicting tx and its descendants as Conflicting and update OutputCache data accordingly
+            let mut tx_to_rollback_data = vec![];
+            for tx_id in txs_to_rollback.iter().rev().copied() {
+                match self.txs.entry(tx_id.into()) {
+                    Entry::Occupied(mut entry) => match entry.get_mut() {
+                        WalletTx::Block(_) => {
+                            Err(WalletError::TransactionIdCannotMapToBlock(tx_id))
+                        }
+                        WalletTx::Tx(tx) => match tx.state() {
+                            TxState::Inactive(_) | TxState::InMempool(_) => {
+                                tx.set_state(TxState::Conflicted(block_id));
+                                tx_to_rollback_data.push(entry.get().clone());
+                                Ok(())
+                            }
+                            TxState::Conflicted(..)
+                            | TxState::Abandoned
+                            | TxState::Confirmed(..) => {
+                                Err(WalletError::CannotMarkTxAsConflictedIfInState(*tx.state()))
+                            }
+                        },
+                    },
+                    Entry::Vacant(_) => {
+                        Err(WalletError::CannotFindDescendantTransactionWithId(tx_id))
+                    }
+                }?;
+            }
+
+            for tx in tx_to_rollback_data {
+                self.rollback_tx_data(&tx)?;
+            }
+
+            conflicting_txs_with_descendants.extend(txs_to_rollback.into_iter());
+        }
+
+        Ok(conflicting_txs_with_descendants)
     }
 
-    fn uses_token(&self, unconfirmed_tx: &WalletTx, frozen_token_id: &TokenId) -> bool {
+    fn violates_frozen_token(&self, unconfirmed_tx: &WalletTx, frozen_token_id: &TokenId) -> bool {
```

Comment:
Btw, probably this should check tx outputs too?

Reply:
Not sure that it's possible to have tokens in the outputs without inputs